ddat_assignment/Areas/Identity/Pages/Account/Manage/CustomerProfile.cshtml.cs

```diff
@@ -0,0 +1,294 @@
+            {
+                var token = await _userManager.GeneratePasswordResetTokenAsync(user);
+                var passwordResult = await _userManager.ResetPasswordAsync(user, token, CustomerPersonalInfo.Password);
+
+                if (!passwordResult.Succeeded)
+                {
+                    foreach (var error in passwordResult.Errors)
+                    {
+                        _logger.LogError("Error updating password: {Error}", error.Description);
+                    }
+                    StatusMessage = "Error updating password.";
+                    return RedirectToPage();
+                }
+            }
+
+            await _signInManager.RefreshSignInAsync(user);
+            StatusMessage = "Your profile has been updated";
+            return RedirectToPage();
+        }
+
+        /*public async Task<IActionResult> OnPostUpdateAddressAsync()
+        {
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+            }
+
+            var userDetails = await _context.UserDetailsModel.FirstOrDefaultAsync(u => u.UserId == user.Id);
+            ModelState.Clear();
+            TryValidateModel(Address, nameof(Address));
+
+            if (!ModelState.IsValid)
+            {
+                await LoadAddressAsync(userDetails);
+                return Page();
+            }
+
+            userDetails.Address = $"{Address.HouseNumber}, {Address.Street}, {Address.Postcode}, {Address.Park}, {Address.City}";
+
+            var updateResult = await _userManager.UpdateAsync(user);
+            if (!updateResult.Succeeded)
+            {
+                foreach (var error in updateResult.Errors)
+                {
+                    _logger.LogError("Error updating address: {Error}", error.Description);
+                }
+                StatusMessage = "Unexpected error when trying to update address.";
+                return RedirectToPage();
+            }
+
+            StatusMessage = "Your address has been updated";
+            return RedirectToPage();
+        }*/
+
+        public async Task<IActionResult> OnPostUpdateAddressAsync()
+        {
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+            }
+
+            var userDetails = await _context.UserDetailsModel.FirstOrDefaultAsync(u => u.UserId == user.Id);
+            ModelState.Clear();
+            TryValidateModel(Address, nameof(Address));
+
+            if (!ModelState.IsValid)
+            {
+                await LoadAddressAsync(userDetails);
+                return Page();
+            }
+
+            userDetails.Address = $"{Address.Address1}, {Address.Address2}, {Address.Postcode}, {Address.City}, {Address.State}";
```

Comment:
![image](https://github.com/yannisleng/ddac_assignment/assets/92348692/8991791e-ce5b-42e5-9f65-2f1f4b900d79)
Replace the "," with "||"; since when you input your address data where contains a comma inside any address (eg "42, Jalan Technology"), using "," will make your data string got extra comma, and when you split t will get additional index.